chartfx-chart/src/main/java/de/gsi/chart/plugins/TableViewer.java

```diff
@@ -257,28 +271,40 @@ private ColumnType(final int dimIdx, final String label, final boolean errorCol,
      * @author akrimm
      */
     protected class DataSetsModel extends ObservableListBase<DataSetsRow> {
+        static final double DEFAULT_COL_WIDTH = 150;
         private int nRows;
         private final ObservableList<TableColumn<DataSetsRow, ?>> columns = FXCollections.observableArrayList();
-        private Callable<Void> refreshFunction;
 
         private long lastColumnUpdate = 0;
         private AtomicBoolean columnUpdateScheduled = new AtomicBoolean(false);
 
         private final ListChangeListener<Renderer> rendererChangeListener = this::rendererChanged;
         private final InvalidationListener datasetChangeListener = this::datasetsChanged;
         private final EventListener dataSetDataUpdateListener = (UpdateEvent evt) -> FXUtils.runFX(() -> this.datasetsChanged(null));
+        private TimerTask timerTask;
 
         public DataSetsModel() {
             super();
             columns.add(new RowIndexHeaderTableColumn());
+            table.visibleProperty().addListener((prop, oldVal, newVal) -> {
+                if (newVal == true) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary comparisons in boolean expressions](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4989692)